dmoj/problem_schema.yml

```diff
@@ -0,0 +1,142 @@
+#normal names
+type: map
+mapping:
+  "archive":
+    type: str
+    pattern: .+\.zip
+  "generator":
+    type: str
+  "checker":
+    type: any
+  "signature_grader":
+    type: map
+    mapping:
+      "entry":
+        type: str
+      "header":
+        type: str
+  "custom_judge":
+    type: str
+  "io_redirects":
+    type: map
+    mapping:
+      "DATA.in":
+        type: map
+        mapping:
+          "fd":
+            type: str
+          "mode":
+            type: str
+      "DATA.out":
+        type: map
+        mapping:
+          "fd":
+            type: str
+          "mode":
+            type: str
+  "hints":
+    type: seq
+    sequence:
+      - type: str
+
+#defaults
+  "wall_time_factor":
+    type: int
+  "output_prefix_length":
+    type: int
+  "output_limit_length":
+    type: int
+
+#names with ++
+  "archive+":
+    type: str
+  "archive++":
+    type: str
+  "generator+":
+    type: str
+  "generator++":
+    type: str
+  "checker+":
+    type: str
+  "checker++":
+    type: str
+  "signature_grader+":
+    type: str
+  "signature_grader++":
+    type: str
+  "custom_judge+":
+    type: str
+  "custom_judge++":
+    type: str
+  "io_redirects+":
+    type: str
+  "io_redirects++":
+    type: str
+  "wall_time_factor+":
+    type: str
+  "wall_time_factor++":
+    type: str
+  "output_prefix_length+":
+    type: str
+  "output_prefix_length++":
+    type: str
+  "output_limit_length+":
+    type: str
+  "output_limit_length++":
+    type: str
+
+# outer points, in, out
+  "points":
+    type: int
+  "in":
+    type: str
+  "out":
+    type: str
+
+# pretest_test_cases
+  "pretest_test_cases":
```

Comment:
This actually has the exact same format as `test_cases`. You could probably group them, maybe.